scripts/west_commands/zcmake.py

```diff
@@ -71,17 +71,20 @@ def run_cmake(args, cwd=None, capture_output=False, dry_run=False):
         raise subprocess.CalledProcessError(p.returncode, p.args)
 
 
-def run_build(build_directory, **kwargs):
+def run_build(build_directory, target=None, **kwargs):
```

Comment:
This is a little tricky in my opinion. Maybe we can just make run_build use extra_args instead the way build.py does? My concern is things like how extra_args in build.py adds additional arguments to the underlying build tool with `--`. If a caller did that *and* set the `target` kwarg, it would produce unexpected and incorrect results.

Not a blocker, just want to explore this alternative instead.

Reply:
I did consider to use the `extra_args` but I felt it was cleaner to have a dedicated parameter for this, and have the `extra_args` reserved for the command line.

> If a caller did that and set the target kwarg

Doing so could already today give unintended side effects, like if you do:
```
west flash -- --target menuconfig
```
that would not build the hex file.

In order to ensure (better) consistency between `west flash` and `ninja flash` (as well as other runners), I think it makes sense to reserve `--target <target>` when invoking `west flash`, as this PR does.

If user wants to build a dedicated target first and NOT have the flash deps target executed, then user can still do:
```
west build --target <mytarget>
west flash --skip-rebuild
```

Reply:
> Doing so could already today give unintended side effects, like if you do:
> 
> ```
> west flash -- --target menuconfig
> ```
> 
> that would not build the hex file.

I don't understand; `west flash` doesn't take extra command line arguments like this, does it?

Reply:
seems you're right, I though the `--extra-args` was for the command line of `west flash` being passed as kwarg when using `--`.

I see that I was wrong on that interpretation, and thus I completely agree, the kwarg should be used instead.

Reply:
fixed.